x/staking/keeper/val_state_change.go

```diff
@@ -306,15 +307,19 @@ func (k Keeper) bondValidator(ctx sdk.Context, validator types.Validator) (types
 	// delete the validator by power index, as the key will change
 	k.DeleteValidatorByPowerIndex(ctx, validator)
 
+	// delete from queue if present
+	k.DeleteValidatorQueue(ctx, validator, validator.UnbondingTime, validator.UnbondingHeight)
+
 	validator = validator.UpdateStatus(types.Bonded)
+	validator.UnbondingHeight = 0
+	validator.UnbondingTime = time.Time{}
+	validator.UnbondingOnHold = false
```

Comment:
This should be moved to `unbondingToBonded`, since it's the only case where `validator.UnbondingOnHold` could be `true` without being set to `false` by the CCV module (via the `validatorUnbondingCanComplete` function).